Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,369 @@
+      apply (snorm_indicator_le _).trans
+      convert hfε.le
+    · have hi' := Fin.is_lt i
+      rw [Nat.lt_succ_iff] at hi'
+      rw [not_lt] at hi
+      -- Porting note: Original proof was `simp [← le_antisymm hi' hi]`
+      ext; symm; rw [Fin.coe_ofNat_eq_mod, le_antisymm hi' hi, Nat.mod_succ_eq_iff_lt, Nat.lt_succ]
+
+/-- A finite sequence of Lp functions is uniformly tight. -/
+theorem unifTight_finite [Finite ι] (hp_top : p ≠ ∞) {f : ι → α → β}
+    (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := fun ε hε ↦ by
+  obtain ⟨n, hn⟩ := Finite.exists_equiv_fin ι
+  set g : Fin n → α → β := f ∘ hn.some.symm
+  have hg : ∀ i, Memℒp (g i) p μ := fun _ => hf _
+  obtain ⟨s, hμs, hfε⟩ := unifTight_fin hp_top hg hε
+  refine ⟨s, hμs, fun i => ?_⟩
+  specialize hfε (hn.some i)
+  unfold_let g at hfε
+  simp_rw [Function.comp_apply, Equiv.symm_apply_apply] at hfε
+  assumption
+
+end UnifTight
+
+
+section VitaliConvergence
+
+variable {μ : Measure α} {p : ℝ≥0∞}
+
+variable {f : ℕ → α → β} {g : α → β}
+
+/-! Both directions and an iff version of Vitali's convergence theorem on measure spaces
+   of not necesserily finite volume. See `Thm III.6.15` of Dunford & Schwartz, Part I (1958). -/
+
+/- We start with the reverse direction. We only need to show that uniform tightness follows
+   from convergence in Lp. Mathlib already has the analogous `unifIntegrable_of_tendsto_Lp`
+   and `tendstoInMeasure_of_tendsto_snorm`. -/
+
+/-- Intermediate lemma for `unifTight_of_tendsto_Lp`. -/
+theorem unifTight_of_tendsto_Lp_zero (hp' : p ≠ ∞) (hf : ∀ n, Memℒp (f n) p μ)
+    (hf_tendsto : Tendsto (fun n => snorm (f n) p μ) atTop (𝓝 0)) : UnifTight f p μ := fun ε hε ↦by
+  rw [ENNReal.tendsto_atTop_zero] at hf_tendsto
+  obtain ⟨N, hNε⟩ := hf_tendsto ε (by simpa only [gt_iff_lt, ofReal_pos])
+  let F : Fin N → α → β := fun n => f n
+  have hF : ∀ n, Memℒp (F n) p μ := fun n => hf n
+  obtain ⟨s, hμs, hFε⟩ := unifTight_fin hp' hF hε
+  refine ⟨s, hμs, fun n => ?_⟩
+  by_cases hn : n < N
+  · exact hFε ⟨n, hn⟩
+  · exact (snorm_indicator_le _).trans (hNε n (not_lt.mp hn))
+
+/-- Convergence in Lp implies uniform tightness. -/
+theorem unifTight_of_tendsto_Lp (hp' : p ≠ ∞) (hf : ∀ n, Memℒp (f n) p μ)
+    (hg : Memℒp g p μ) (hfg : Tendsto (fun n => snorm (f n - g) p μ) atTop (𝓝 0)) :
+    UnifTight f p μ := by
+  have : f = (fun _ => g) + fun n => f n - g := by ext1 n; simp
+  rw [this]
+  refine UnifTight.add ?_ ?_ (fun _ => hg.aestronglyMeasurable)
+      fun n => (hf n).1.sub hg.aestronglyMeasurable
+  · exact unifTight_const hp' hg
+  · exact unifTight_of_tendsto_Lp_zero hp' (fun n => (hf n).sub hg) hfg
+
+
+/- Next we deal with the forward direction. The `Memℒp` and `TendstoInMeasure` hypotheses
+   are unwrapped and strengthened to by known lemmas to have in addition `StronglyMeasurable`
+   and a.e. convergence. The bulk of the proof is done under these stronger hyptheses. -/
+
+theorem tendsto_Lp_notFinite_of_tendsto_ae_of_meas (hp : 1 ≤ p) (hp' : p ≠ ∞)
```

Comment:
Uh, 
```suggestion
theorem tendsto_Lp_infinite_of_tendsto_ae_of_meas (hp : 1 ≤ p) (hp' : p ≠ ∞)
```
?
But there is no “infinite” in the statement so maybe
```suggestion
theorem tendsto_Lp_of_tendsto_ae_of_meas (hp : 1 ≤ p) (hp' : p ≠ ∞)
```

Reply:
Could you make the change in a *preliminary PR*? It will be strictly better in the history to see
* rename `B` to `C`
* introduce `B`

rather than
* introduce `A`
* rename `B` to `C` and `A` to `B`